source/documentation/monitoring-an-app/application-metrics.md

```diff
@@ -1,107 +1,121 @@
+```
+gem `prometheus-client`
+```
 
-See screenshot below of how the metrics endpoint looks on a browser:
-![Image of metrics](https://raw.githubusercontent.com/ministryofjustice/cloud-platform-user-docs/master/images/metrics-endpoint.png)
+Next we need to amend the `config.ru` file and inclued the two `rack` middlewares required by the `prometheus-client`. 
 
+```
+require_relative 'config/environment'
+require 'prometheus/middleware/collector'
+require 'prometheus/middleware/exporter'
 
-#### Create a Service to expose Pods
+use Prometheus::Middleware::Collector
+use Prometheus::Middleware::Exporter
 
-Scraping an exporter or separate metrics port requires a service that targets the Pod(s) of the exporter or application.
+run Rails.application
+```
 
-Example:
+If you're running this locally, you'll now be able to query your `/metrics` endpoint and display an output. 
 
-```yaml
-kind: Service
-apiVersion: v1
-metadata:
-  name: my-app-service
-  labels:
-    app: my-app
-spec:
-  selector:
-    app: my-app
-  ports:
-  - protocol: TCP
-    port: 8000
-    targetPort: 8000
-    name: metrics
-  type: NodePort
+```
+curl localhost:3000/metrics
 ```
 
-#### Create a Service-Monitor
-A `ServiceMonitor` is a resource the Prometheus Operator introduces for Kubernetes that describes the set of targets to be monitored by Prometheus
+Build, tag and push your application changes to your code repository and deploy the latest version into your Cloud Platform namespace. Confirm your `/metrics` endpoint is now accessible from your url. 
 
-Service Monitor Architecture
-![Image of Service-Monitor Architecture](https://raw.githubusercontent.com/ministryofjustice/cloud-platform-user-docs/master/images/service-monitor-arch.png)
+```
+curl https://myapp.cloud-platform/metrics
+```
 
+#### Add Service endpoint and ServiceMonitor
 
-Example:
+We need to expose the metrics endpoint with a `Service` and tell the Cloud Platform Prometheus to scrape the endpoint with `ServiceMonitor` object in Kubernetes. In this example, we're using the same port to expose both our application and metrics endpoint so we'll need to query our existing `Service` for the current port name and number. However, if you're exposing a different port you'll need to either amend your current `Service` or create a new one. 
 
-```yaml
-apiVersion: monitoring.coreos.com/v1
-kind: ServiceMonitor
-metadata:
-  name: my-app
-spec:
-  selector:
-    matchLabels:
-      app: my-app
-  endpoints:
-  - port: metrics
-    interval: 15s
-```
+Let's find out our current port name and number by running:
 
-More detailed information about service-monitors can be found [here](https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/running-exporters.md)
+```
+kubectl get svc -n <namespace>
+```
 
```

Comment:
What does the output look like, and how does the developer make use of it?

Reply:
Amended the command to pull the port name only.